extensions/rocksdb-repos/ProvenanceRepository.h

```diff
@@ -156,7 +162,7 @@ class ProvenanceRepository : public core::Repository, public std::enable_shared_
       std::string key = it->key().ToString();
       if (store.size() >= max_size)
         break;
-      if (eventRead->DeSerialize((uint8_t *) it->value().data(), (int) it->value().size())) {
+      if (eventRead->DeSerialize(reinterpret_cast<const uint8_t *>(it->value().data()), static_cast<int>(it->value().size()))) {
```

Comment:
this cast to `int` looks dangerous (we may be silently dropping the high-order bits), and it is also unnecessary, as the second parameter of `DeSerialize()` is of type `size_t` [also in a few places further down in this file]
```suggestion
      if (eventRead->DeSerialize(reinterpret_cast<const uint8_t *>(it->value().data()), it->value().size())) {
```